backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java

```diff
@@ -280,12 +300,101 @@ public String getUrlForDatasetVersion(String datasetVersionId, String method)
     }
   }
 
+  /**
+   * For getting datasetVersions that user has access to (either as owner or a collaborator), fetch
+   * all datasetVersions of the requested datasetVersionIds then iterate that list and check if
+   * datasetVersion is accessible or not. The list of accessible datasetVersionIDs is built and
+   * returned by this method.
```

Comment:
This description doesn't include `dataset`, though it's used in the text. I think you mean it in a couple of places that you mention `datasetVersion`.